server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -325,7 +332,20 @@ private void addSequencesFromServer(ArrayList<Pair<Interval, Sequence<T>>> listO
               final DruidServer server = entry.getKey();
               final List<SegmentDescriptor> descriptors = entry.getValue();
 
-              final QueryRunner clientQueryable = serverView.getQueryRunner(server);
+              final QueryRunner clientQueryable = new MetricsEmittingQueryRunner(
+                  emitter,
+                  new Function<Query<T>, ServiceMetricEvent.Builder>()
+                  {
+                    @Override
+                    public ServiceMetricEvent.Builder apply(@Nullable final Query<T> input)
+                    {
+                      return toolChest.makeMetricBuilder(input);
+                    }
+                  },
+                  serverView.getQueryRunner(server),
+                  "query/node/time"
```

Comment:
this is a good metric to have but it will be even more useful if metric event includes the name of historical/realtime node as a dimension. so, when I see high/abnormal value of this metric, I would immediately know which historical/realtime node is causing trouble.